logger/honeycomb.go

```diff
@@ -90,6 +95,24 @@ func (h *HoneycombLogger) Stop() error {
 	return nil
 }
 
+func (h *HoneycombLogger) WithField(key string, value interface{}) Entry {
+	entry := &HoneycombEntry{
+		loggerConfig: h.loggerConfig,
+		builder:      h.builder.Clone(),
+	}
+	entry.builder.AddField(key, value)
+	return entry
+}
+
+func (h *HoneycombLogger) WithFields(fields map[string]interface{}) Entry {
+	entry := &HoneycombEntry{
+		loggerConfig: h.loggerConfig,
+		builder:      h.builder.Clone(),
```

Comment:
once we have an entry, couldn't we use an `Event` rather than cloning the builder with each chain?

Reply:
Just thinking about GC on all of these builders. I guess we also have this problem with events too, since we create an event even if it never will be sent based on the log level.

Reply:
done, reuse builder once it's created to an Entry.